STYLING.md

```diff
@@ -0,0 +1,216 @@
+# React Native Styling Guidelines
+
+## Where to Define Styles
+
+All styles must be defined in the `/styles` directory and `styles.js` contains the final export after gathering all appropriate styles. Unlike some React Native applications we are not using `StyleSheet.create()` and instead store styles as plain JS objects. There are also many helper styles available for direct use in components.
+
+These helper styles are loosely based on the [Bootstrap system of CSS utility helper classes](https://getbootstrap.com/docs/5.0/utilities/spacing/) and are typically incremented by units of `4`.
+
+**Note:** Not all helpers from Bootstrap exist, so it may be necessary to create the helper style we need.
+
+## When to Create a New Style
+
+If we need some minimal set of styling rules applied to a single-use component then it's almost always better to use an array of helper styles rather than create an entirely new style if it will only be used once. Resist the urge to create a new style for every new element added to a screen. There is a very good chance the style we are adding is a "single-use" style.
+
+```jsx
+// Bad - Since we only use this style once in this component
+const TextWithPadding = props => (
+    <Text style={styles.textWithPadding}>
+        {props.children}
+    </Text>
+);
+
+// Good
+const TextWithPadding = props => (
+    <Text
+        style={[
+            styles.p5,
+            styles.noWrap,
+        ]}
+    >
+        {props.children}
+    </Text>
+);
+```
+
+On the other hand, if we are copying and pasting some chunks of JSX from one place to another then that might be a sign that we need a new reusable style.
+
+## Use the "Rule of Three"
+
+In order to resist the urge to preoptimize and have many single-use components we've adopted a main principle:
+
+Any array of styles associated with a single type of React element that has at least 3 identical usages should be refactored into:
+
+- A new resusable style that can be used in many places e.g. `styles.button`
+- If that style has modifiers or style variations then those styles should follow a naming convention of `styles.elementModifer` e.g. `styles.buttonSuccess`
+- If a reusable style has 3 or more modifiers it should be refactored into a component with props to modify the styles e.g.
+
+```jsx
+<Button title="Submit" success large />
+```
+
+## Inline Styles
+
+**Inline styles are forbidden.** If we run into a case where we feel it's necessary to conditionally render some styles we should create a helper function and import it into `styles.js` then pass any modifying parameters to that function e.g.
```

Comment:
> create a helper function and import it into `styles.js`

So we decided to have dynamic style generator functions live in separate files, like `getModalStyles.js`, instead of putting those functions directly in `styles.js`? It's fine with me, just wanted to clarify that this is the intention.

Reply:
I think they can live directly in `styles.js` unless they are very large and there is some necessity to create a new file? But don't really have strong feelings on this point.

Reply:
Cool, I agree. Maybe we can add a sentence like:

"Small helper functions can be written directly in `styles.js`, but larger, more complex dynamic style generators can be put in their own modules"